pkg/usersignup/usersignup.go

```diff
@@ -30,5 +37,40 @@ func TransformUsername(username string) string {
 	if strings.HasSuffix(newUsername, "-") {
 		newUsername = newUsername + "crt"
 	}
+	if len(newUsername) > MaxLength {
+		newUsername = newUsername[0:MaxLength]
+		if len(validation.IsDNS1123Label(newUsername)) != 0 {
+			// trim starting or ending hyphen
+			newUsername = strings.Trim(newUsername, "-")
+		}
+	}
+
+	// -4 for "crt-" or "-crt" to be added in following lines. Update this length changing prefix
+	maxlengthWithPrefix := MaxLength - 4
+	// Check for any forbidden prefixes
+	for _, prefix := range ForbiddenUsernamePrefixes {
+		if strings.HasPrefix(newUsername, prefix) {
+			if len(newUsername) > maxlengthWithPrefix {
+				// replace prefix instead of append
+				newUsername = "crt-" + newUsername[3:]
+			} else {
+				newUsername = fmt.Sprintf("%s%s", "crt-", newUsername)
+			}
+			break
+		}
+	}
+
+	// Check for any forbidden suffixes
+	for _, suffix := range ForbiddenUsernameSuffixes {
+		if strings.HasSuffix(newUsername, suffix) {
+			if len(newUsername) > maxlengthWithPrefix {
+				// replace prefix instead of append
+				newUsername = newUsername[:len(newUsername)-4] + "-crt"
```

Comment:
just out of curiosity - could it be also:
```suggestion
				newUsername = newUsername[:maxlengthWithPrefix] + "-crt"
```
so we would reuse the variable with the expected length that is already set and used before?